applications/registration-react/src/pages/dataset-list-page/dataset-list-page.jsx

```diff
@@ -1,91 +1,72 @@
 import React from 'react';
-import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import _ from 'lodash';
 
-import {
-  fetchHelptextsIfNeeded,
-  fetchCatalogIfNeeded
-} from '../../actions/index';
-import {
-  fetchDatasetsIfNeeded,
-  getDatasetItemsByCatalogId
-} from '../../redux/modules/datasets';
+import { getDatasetItemsByCatalogId } from '../../redux/modules/datasets';
 import FormCatalog from './form-catalog/connected-form-catalog.component';
 import DatasetItemsList from './items-list/item-list.component';
 import './dataset-list-page.scss';
 
-export class RegDatasetsList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-    const catalogURL = window.location.pathname;
-    const catalogId = catalogURL.split('/')[2];
-    this.props.dispatch(fetchHelptextsIfNeeded());
-    this.props.dispatch(fetchCatalogIfNeeded(`/catalogs/${catalogId}`));
-    this.props.dispatch(fetchDatasetsIfNeeded(catalogId));
+export const DatasetsListPage = props => {
+  const {
+    helptexts,
+    catalog,
+    datasets,
+    fetchHelptextsIfNeeded,
+    fetchCatalogIfNeeded,
+    fetchDatasetsIfNeeded,
+    match
+  } = props;
+
+  const catalogId = _.get(match, ['params', 'catalogId']);
+  if (catalogId) {
+    fetchCatalogIfNeeded(catalogId);
+    fetchDatasetsIfNeeded(catalogId);
   }
 
-  render() {
-    const {
-      helptextItems,
-      isFetchingCatalog,
-      catalogItem,
-      datasets
-    } = this.props;
-    return (
-      <div className="container">
-        <div className="row">
-          <div className="col-12" />
-        </div>
-        <div className="row mb-2 mb-md-5">
-          {helptextItems &&
-            !isFetchingCatalog &&
-            catalogItem && (
-              <div className="col-12 fdk-reg-datasets-list">
-                <FormCatalog helptextItems={helptextItems} />
-                <DatasetItemsList
-                  catalogId={catalogItem.id}
-                  datasetItems={getDatasetItemsByCatalogId(
-                    datasets,
-                    catalogItem.id
-                  )}
-                />
-              </div>
-            )}
-        </div>
+  fetchHelptextsIfNeeded();
+
+  return (
+    <div className="container">
+      <div className="row">
+        <div className="col-12" />
       </div>
-    );
-  }
-}
+      <div className="row mb-2 mb-md-5">
+        {_.get(helptexts, 'helptextItems') &&
+          _.get(catalog, ['items', catalogId]) &&
```

Comment:
når vi tenker om keyetstore lib, må tenke litt, for di det er litt forsvirrende. at noe av store, har element som data, men andre har array som data... for mye "items" og de mener forkjellige ting.